Add new jobs for jenkins

cosmin-malutan opened the pull request:
This is for issue #24, I added just jobs for desktop, for mobile if we decide we want to run them to we can just copy the job and add an extra parameter in xshell command. 
Regarding trigger.py, I added just a  plain script with options so it won't throw when called, this will be overwrited by upcoming pr from issue #17.

cosmin-malutan committed:
Add new jobs for jenkins (#24)

cosmin-malutan commented:
@whimboo can you have a look at this?

whimboo commented:
Here some comments while looking at this:
- I don't think we have to focus that much on mobile right now, given that those tests are run on desktop right now. Once we have real mobile devices it will look much differently.
- As I was thinking over the weekend, we probably cannot run tests via the old sync account data, given that there is AFAIK no Python client, which let us setup and remove those accounts. But lets ask. 
- I don't like the name of the jobs. While the prefix is fine, the remaining part is not telling us much. Think about which other kind of tests we might have to run here in the future.
- The `tools` job  I would rename to `scripts`, so it indicates that it holds scripts, but no external tools.

More inline.

whimboo commented on jenkins-master/config.xml:
```diff
@@ -39,6 +39,119 @@
     </hudson.model.AllView>
+    <listView>
+      <owner class="hudson" reference="../../.."/>
+      <name>-mozilla-central</name>
```
Please keep the alphabetical order for views.

whimboo commented on jenkins-master/config.xml:
```diff
@@ -39,6 +39,119 @@
+    </listView>
+    <listView>
+      <owner class="hudson" reference="../../.."/>
+      <name>-mozilla-beta</name>
```
There are no nightly builds for beta

cosmin-malutan replied:
I removed those, I understood from this that I should remove it. If I should just added "-release" tag to the name of view, I will do that.

whimboo replied:
I never told that those jobs have to be removed. I just said we do not have Nightly builds for beta and releases. So please read the comments more carefully, and make some more thoughts before jumping on doing the changes. Also if something is unclear I still miss questions.

cosmin-malutan replied:
I'll add those back, and add -release prefix for the views.
Regarding job suffix I will change it to fx-account, so the release job for example it will be:
`release-mozilla-release_fx-account`

whimboo replied:
I think that's fine.

whimboo commented on jenkins-master/config.xml:
```diff
@@ -39,6 +39,119 @@
+    </listView>
+    <listView>
+      <owner class="hudson" reference="../../.."/>
+      <name>-mozilla-release</name>
```
There are no nightly builds for releases.

whimboo commented on jenkins-master/jobs/mozilla-aurora_desktop/config.xml:
```diff
@@ -0,0 +1,87 @@
+  <description>Execute desktop TPS tests for Aurora builds.</description>
+  <logRotator class="hudson.tasks.LogRotator">
+    <daysToKeep>-1</daysToKeep>
+    <numToKeep>-1</numToKeep>
```
I mentioned more than once that we want to keep as much as possible from the mozmill-ci project. Something we totally need is log rotation. Do you want to manually update the master from old work spaces? This is an essential feature, which is missing.

whimboo commented on jenkins-master/jobs/mozilla-aurora_desktop/config.xml:
```diff
@@ -0,0 +1,87 @@
+    <hudson.model.ParametersDefinitionProperty>
+      <parameterDefinitions>
+        <hudson.model.StringParameterDefinition>
+          <name>BUILDURL</name>
```
As usual constants with all capital letters get an `_` to separate words.

whimboo commented on jenkins-master/jobs/mozilla-aurora_desktop/config.xml:
```diff
@@ -0,0 +1,87 @@
+      <parameterDefinitions>
+        <hudson.model.StringParameterDefinition>
+          <name>BUILDURL</name>
+          <description>The url path to firefox build.</description>
```
This should read: "The URL to the Firefox build on the FTP server".

whimboo commented on jenkins-master/jobs/mozilla-aurora_desktop/config.xml:
```diff
@@ -0,0 +1,87 @@
+        </hudson.model.StringParameterDefinition>
+        <hudson.model.StringParameterDefinition>
+          <name>TESTSURL</name>
+          <description>The url path to firefox build tests.</description>
```
Same as above.

whimboo commented on jenkins-master/jobs/mozilla-aurora_desktop/config.xml:
```diff
@@ -0,0 +1,87 @@
+          <triggerIfResult>allCases</triggerIfResult>
+          <nodeEligibility class="org.jvnet.jenkins.plugins.nodelabelparameter.node.AllNodeEligibility"/>
+        </org.jvnet.jenkins.plugins.nodelabelparameter.LabelParameterDefinition>
+      </parameterDefinitions>
```
You should think about which other kind of parameters are needed for a proper reporting of Jenkins.

whimboo commented on jenkins-master/jobs/mozilla-aurora_desktop/config.xml:
```diff
@@ -0,0 +1,87 @@
+      <patterns>
+        <hudson.plugins.ws__cleanup.Pattern>
+          <pattern>trigger.py</pattern>
+          <type>EXCLUDE</type>
```
I don't see why this exclusion is necessary.

whimboo commented on jenkins-master/jobs/tools/workspace/trigger.py:
```diff
@@ -0,0 +1,42 @@
+                      type='string',
+                      dest='tests',
+                      default=None,
+                      help='Path to tests zip file, specified as a url')
```
I would use `--testurl` and a help messages correlated to the Jenkins description of the parameter.

whimboo commented on jenkins-master/jobs/tools/workspace/trigger.py:
```diff
@@ -0,0 +1,42 @@
+                      type='string',
+                      dest='build',
+                      default=None,
+                      help='Path to the Firefox binary, specified as a url')
```
Same here as above.

whimboo commented on jenkins-master/jobs/tools/workspace/trigger.py:
```diff
@@ -0,0 +1,42 @@
+                      action='store',
+                      dest='debug',
+                      default='false',
+                      help='run in debug mode')
```
Please check the following URL how to correctly handle boolean values: https://docs.python.org/2/library/optparse.html#handling-boolean-flag-options

cosmin-malutan replied:
I know how to treat boolean values, see `--igonre-unused-engines`, I made this decision because we want to treat the presence of the boolean value from jenkins job variables. Otherwise we would have to append the option in to the xshell command, and we don't have a clean way to do that.
We had this issue on mozmill-automation issue #132 while handling `--override-update-settings` option
 https://github.com/mozilla/mozmill-automation/pull/133/files

whimboo replied:
If that works, it would be fine then.

whimboo commented on jenkins-master/jobs/tools/workspace/trigger.py:
```diff
@@ -0,0 +1,42 @@
+                      action='store_true',
+                      dest='ignore_unused_engines',
+                      help='If defined, do not load unused engines in '
+                           'individual tests. Has no effect for pulse monitor.')
```
I don't think we need this for our CI. Lets keep it removed for now.

whimboo commented:
If we are going with the clone-workspace-scm.jpi plugin, you want to remove the copy_artifact plugin. But first please check any kind of issues with clone-workspace-scm before making this decision.

cosmin-malutan commented:
@whimboo, I dropped the  clone-workspace-scm.jpi plugin and switched to copy_artifact plugin.
Regarding testruns names I renamed it to sync (mozilla-central_sync) at the moment, because I don't know how we could split/change what we test in the future. We could test the ui/backend mobile/desktop FX account/ Sync account, so I thought it would be better to rename them then, if it doesn't looks nice I'm open to any suggestions.

whimboo commented on jenkins-master/config.xml:
```diff
@@ -39,6 +39,72 @@
+        <hudson.views.LastDurationColumn/>
+        <hudson.views.BuildButtonColumn/>
+      </columns>
+      <includeRegex>^mozilla-central.*</includeRegex>
```
Beside the nightly builds I do not see entries for beta and release builds. Where are those?

whimboo commented on jenkins-master/jobs/mozilla-aurora_sync/config.xml:
```diff
@@ -0,0 +1,85 @@
+        </hudson.model.StringParameterDefinition>
+        <hudson.model.StringParameterDefinition>
+          <name>TESTS_URL</name>
+          <description>The URL to the Firefox tests on the FTP server.</description>
```
Please say `to the packaged tests...`.

whimboo commented on jenkins-master/jobs/mozilla-aurora_sync/config.xml:
```diff
@@ -0,0 +1,85 @@
+        </hudson.model.StringParameterDefinition>
+        <hudson.model.BooleanParameterDefinition>
+          <name>DEBUG</name>
+          <description></description>
```
Missing description.

whimboo commented on jenkins-master/jobs/mozilla-aurora_sync/config.xml:
```diff
@@ -0,0 +1,85 @@
+      <doNotFingerprintArtifacts>false</doNotFingerprintArtifacts>
+    </hudson.plugins.copyartifact.CopyArtifact>
+    <hudson.plugins.xshell.XShellBuilder plugin="xshell@0.9">
+      <commandLine>python trigger.py --testurl $TESTS_URL --buildurl $BUILD_URL --debug $DEBUG</commandLine>
```
Maybe we want to use `URL_BUILD` and `URL_TESTS` for the parameter? It helps to directly see that we have URLs here.

whimboo commented:
I haven't said to drop the clone-workspace plugin. I asked you what the benefits are by using it compared to copy artifacts. But you didn't gave any answer to your observation. What's that bad on clone-workspace, so that you have replaced it?

Regarding the job names we will have complete testruns for desktop, which span any backend tests and ui tests driven with Mozmill. There is no distinction yet, and I don't think we will have such in the near future. But separation might exist for the authentication method. That would have to be part of the job name.

cosmin-malutan commented:
I haven't kept the clone-workspace plugin because we already had copy_artifacts landed and it didn't seemed to me a good practice to add a new plugin in favor of already existing one which does basically the same thing.
Beside the fact that copy_artifacts it's more popular I can't see any big difference between them.

cosmin-malutan commented:
@whimboo I made the changes, and I hope I replayed to all questions.

whimboo commented on jenkins-master/jobs/scripts/config.xml:
```diff
@@ -0,0 +1,25 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<project>
+  <actions/>
+  <description>Job for holding the helper scripts for running the TPS testruns </description>
```
I think this script has to be run manually after the first start? In that case please update the README file and mention that.

whimboo commented:
Beside the remaining update for the README it looks fine.

cosmin-malutan commented:
@whimboo I updated the readme, and squashed the commits.

whimboo commented:
Quick note once again... I would prefer if you rebase your commits when the PR has been finalized, and no additional content gets added.

whimboo commented on README.md:
```diff
@@ -22,6 +22,11 @@ can tell when Jenkins is running by looking out for "Jenkins is fully up and
+If this is the first time you've started Jenkins, or your workspaces have
+recently been deleted, you will need to run an admin job to finish the setup.
+Open http://localhost:8080/view/+admin/ and schedule a build for the 'scripts'
+job by clicking the clock icon in the last column.
```
It's enough to say `+admin/, and build the 'scripts' job.`

cosmin-malutan commented:
@whimboo I updated the nit

whimboo commented:
Ok, looks fine. Please rebase the PR, so I can get it landed. Thanks.

cosmin-malutan commented:
@whimboo Rebased, Thanks.
